src/container_runner.py

```diff
@@ -1,38 +1,42 @@
 """Control a OCI image running via Docker on a host system. Provides a ContainerRunner class."""
 
+import json
+from pathlib import Path
 import logging
 import subprocess
 import time
 import os
 from typing import Iterable, Optional
 
-from charms.operator_libs_linux.v1 import snap
-
 # Configure logging level based on environment variable or default to INFO
 log_level = os.getenv("LOG_LEVEL", "WARNING").upper()
 
 # Set up the logger
 logger = logging.getLogger(__name__)
 logger.setLevel(log_level)
 
+DOCKER_DAEMON_CONFIG_PATH = Path("/etc/docker/daemon.json")
+
 
 class _Docker:
     """Private class for handling the installed Docker snap."""
 
-    @property
-    def _docker(self):
-        """Return a representation of the Docker snap."""
-        cache = snap.SnapCache()
-        return cache["docker"]
-
     def install(self):
-        # Install Docker
         try:
-            self._docker.ensure(snap.SnapState.Latest, channel="stable")
-            snap.hold_refresh()
+            # Set this env to automatically restart daemons. If not, the prompt is blocking.
+            env = os.environ.copy()
+            env["NEEDRESTART_MODE"] = "a"
```

Comment:
might be worth linking to the man page on needrestart to explain what the `"a"` value here is doing: https://manpages.ubuntu.com/manpages/jammy/man1/needrestart.1.html